// react-sticky-table/dist/__tests__/index.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["react", "enzyme", "enzyme-adapter-react-16", "chai", "../index"], factory);
  } else if (typeof exports !== "undefined") {
    factory(require("react"), require("enzyme"), require("enzyme-adapter-react-16"), require("chai"), require("../index"));
  } else {
    var mod = {
      exports: {}
    };
    factory(global.react, global.enzyme, global.enzymeAdapterReact16, global.chai, global.index);
    global.index = mod.exports;
  }
})(this, function (_react, _enzyme, _enzymeAdapterReact, _chai, _index) {
  "use strict";

  _react = _interopRequireDefault(_react);
  _enzymeAdapterReact = _interopRequireDefault(_enzymeAdapterReact);

  function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

  (0, _enzyme.configure)({
    adapter: new _enzymeAdapterReact["default"]()
  });
  describe('StickyTable', function () {
    it('should render the correct number of children', function () {
      var table = (0, _enzyme.mount)(_react["default"].createElement(_index.StickyTable, {
        stickyColumnCount: 1
      }, _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'header 1'), _react["default"].createElement(_index.Cell, null, 'header 2')), _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'column 1 cell'), _react["default"].createElement(_index.Cell, null, 'column 2 cell'))));
      (0, _chai.expect)(table.find('.sticky-table-cell')).to.have.length(4);
    });
    it('shouldn\'t render columns when opting out', function () {
      var table = (0, _enzyme.mount)(_react["default"].createElement(_index.StickyTable, {
        stickyColumnCount: 0
      }, _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'header 1'), _react["default"].createElement(_index.Cell, null, 'header 2')), _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'column 1 cell'), _react["default"].createElement(_index.Cell, null, 'column 2 cell'))));
      (0, _chai.expect)(table.find('.sticky-table-cell')).to.have.length(4);
    });
    it('should render the correct number of stickyColumnCount', function () {
      var table = (0, _enzyme.mount)(_react["default"].createElement(_index.StickyTable, {
        stickyColumnCount: 3
      }, _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'header 1'), _react["default"].createElement(_index.Cell, null, 'header 2'), _react["default"].createElement(_index.Cell, null, 'header 3'), _react["default"].createElement(_index.Cell, null, 'header 4')), _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'column 1 cell'), _react["default"].createElement(_index.Cell, null, 'column 2 cell'), _react["default"].createElement(_index.Cell, null, 'column 3 cell'), _react["default"].createElement(_index.Cell, null, 'column 4 cell'))));
      (0, _chai.expect)(table.find('.sticky-table-cell')).to.have.length(8);
    });
    it('shouldn\'t render headers when opting out', function () {
      var table = (0, _enzyme.mount)(_react["default"].createElement(_index.StickyTable, {
        stickyColumnCount: 1,
        stickyHeaderCount: 0
      }, _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'header 1'), _react["default"].createElement(_index.Cell, null, 'header 2')), _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'column 1 cell'), _react["default"].createElement(_index.Cell, null, 'column 2 cell'))));
      (0, _chai.expect)(table.find('.sticky-table-cell')).to.have.length(4);
    });
    it('should support sibling components and arrays', function () {
      var table = (0, _enzyme.mount)(_react["default"].createElement(_index.StickyTable, {
        stickyColumnCount: 1
      }, _react["default"].createElement(_index.Row, null, _react["default"].createElement(_index.Cell, null, 'header 1'), [_react["default"].createElement(_index.Cell, {
        key: 1
      }, 'header 2')])));
      (0, _chai.expect)(table.find('.sticky-table-cell')).to.have.length(2);
    });
  });
});
